echemdb/cv/cv_entry.py

```diff
@@ -0,0 +1,262 @@
+
+        """
+        if field_name in self.package.get_resource("echemdb").schema.field_names:
+            return field_name
+        if field_name == "j":
+            return self._normalize_field_name("I")
+        raise ValueError(f"No axis with name '{field_name}' found.")
+
+    def thumbnail(self, width=96, height=72, dpi=72, **kwds):
+        r"""
+        Return a thumbnail of the entry's curve as a PNG byte stream.
+
+        EXAMPLES::
+
+            >>> entry = CVEntry.create_examples()[0]
+            >>> entry.thumbnail()
+            b'\x89PNG...'
+
+        The PNG's ``width`` and ``height`` can be specified in pixels.
+        Additional keyword arguments are passed to the data frame plotting
+        method::
+
+            >>> entry.thumbnail(width=4, height=2, color='red', linewidth=2)
+            b"\x89PNG..."
+
+        """
+        kwds.setdefault("color", "b")
+        kwds.setdefault("linewidth", 1)
+        kwds.setdefault("legend", False)
+
+        import matplotlib.pyplot
+
+        # A reasonable DPI setting that should work for most screens is the default value of 72.
+        fig, axis = matplotlib.pyplot.subplots(
+            1, 1, figsize=[width / dpi, height / dpi], dpi=dpi
+        )
+        self.df.plot(
+            "E",
+            self._normalize_field_name("j"),
+            ax=axis,
+            **kwds,
+        )
+
+        matplotlib.pyplot.axis("off")
+        matplotlib.pyplot.close(fig)
+
+        import io
+
+        buffer = io.BytesIO()
+        fig.savefig(buffer, format="png", transparent=True, dpi=dpi)
+
+        buffer.seek(0)
+        return buffer.read()
+
+    def plot(self, x_label="E", y_label="j"):
+        r"""
+        Return a plot of this entry.
+        The default plot is a cyclic voltammogram ('j vs E').
+        When `j` is not defined `I` is used instead.
+
+        EXAMPLES::
+
+            >>> entry = CVEntry.create_examples()[0]
+            >>> entry.plot()
+            Figure(...)
+
+        The plot can also be returned with custom axis units available in the resource::
+
+            >>> entry.plot(x_label='t', y_label='E')
+            Figure(...)
+
+        The plot with axis units of the original figure can be obtained by first rescaling the entry::
+
+            >>> rescaled_entry = entry.rescale('original')
+            >>> rescaled_entry.plot()
+            Figure(...)
+
+        """
+        import plotly.graph_objects
+
+        x_label = self._normalize_field_name(x_label)
+        y_label = self._normalize_field_name(y_label)
+
+        fig = plotly.graph_objects.Figure()
+
+        def figure_name():
+            try:
+                self.source.figure
+            except AttributeError:
+                try:
+                    self.source.curve
+                except AttributeError:
+                    return self.identifier
+            return f"Fig. {self.source.figure}: {self.source.curve}"
+
+        # pylint: disable=duplicate-code
```

Comment:
Without having thought about this much, I agree with pylint that this looks like it's duplicating `Entry.plot`.

Reply:
So, I think we want to refactor this so that the functionality is in the base class and the subclass just calls the base class in some specialized way.

Reply:
I was thinking about that. But maybe it is more intuitive this way to understand how the plot is created and how it is customized. In principle `fig.add_trace(...)` is identical in both classes `entry` and `cv_entry`, while `fig.update_layout` involves minor changes. In between, there is a bunch of code specific to the dataset. But let's discuss this briefly.